src/proxy/ddl_test.go

```diff
@@ -540,17 +540,82 @@ func TestProxyDDLConstraint(t *testing.T) {
+		"create table t7(a int unique key, b int key)   PARTITION  BY hash(a)  ",
+		"create table t8(a int primary key, b int key) partition by hash(a)",
+		"create table t9(a int, b int, primary key(a))engine=tokudb PARTITION  BY hash(a)  ",
+		"create table t10(a int key, b int, primary key(a))engine=tokudb PARTITION  BY hash(a)  ",
+		"create table t11(a int key, b int key, primary key(a))engine=tokudb PARTITION  BY hash(a)  ",
+		"create table t12(a int, b int, primary key(a,b)) default charset=utf8  PARTITION  BY hash(a)  ",
+		"create table t13(a int key, b int, primary key(a,b)) default charset=utf8  PARTITION  BY hash(a)  ",
+		"create table t14(a int key, b int key, primary key(a,b)) default charset=utf8  PARTITION  BY hash(a)  ",
+		"create table t15(a int unique, b int, primary key(a,b))engine=tokudb default charset=utf8  PARTITION  BY hash(a)  ",
+		"create table t16(a int unique key, b int key, primary key(a,b))engine=tokudb default charset=utf8  PARTITION  BY hash(a)  ",
+		"create table t17(a int unique, b int key, primary key(a))engine=tokudb default charset=utf8  PARTITION  BY hash(a)  ",
+		"create table t18(a int unique, b int key, key `name` (`a`))engine=tokudb default charset=utf8  PARTITION  BY hash(a)  ",
+		"create table t19(a int unique, b int key, index `name` (a))engine=tokudb default charset=utf8  PARTITION  BY hash(a)  ",
+		"create table t20(a int unique, b int key, unique index `name` (a))engine=tokudb default charset=utf8  PARTITION  BY hash(a)  ",
+		"create table t21(a int unique, b int key, unique key `name` (a))engine=tokudb default charset=utf8  PARTITION  BY hash(a)  ",
+	}
+
+	for _, query := range querys {
+		client, err := driver.NewConn("mock", "mock", address, "test", "utf8")
+		assert.Nil(t, err)
+		_, err = client.FetchAll(query, -1)
+		assert.Nil(t, err)
+	}
+}
+
+func TestProxyDDLConstraintError(t *testing.T) {
+	log := xlog.NewStdLog(xlog.Level(xlog.PANIC))
+	fakedbs, proxy, cleanup := MockProxy(log)
+	defer cleanup()
+	address := proxy.Address()
+
+	// fakedbs.
+	{
+		fakedbs.AddQueryPattern("use .*", &sqltypes.Result{})
+		fakedbs.AddQueryPattern("create table .*", &sqltypes.Result{})
+	}
+
+	querys := []string{
+		"create table t1(a int unique index, b int unique) partition by hash(a)",
+		"create table t2(a int, b int unique) partition by hash(a)",
+		"create table t3(a int unique, b int unique) partition by hash(a)",
+		"create table t4(a int, b int primary key) PARTITION BY hash(a)",
+		"create table t5(a int unique key, b int primary key)   PARTITION  BY hash(a)  ",
+		"create table t6(a int primary key, b int primary key) partition by hash(a)",
+		"create table t7(a int, b int unique, primary key(a))engine=tokudb PARTITION  BY hash(a)  ",
+		"create table t8(a int, b int unique key, primary key(a))engine=tokudb PARTITION  BY hash(a)  ",
+		"create table t9(a int key, b int key, primary key(a))engine=tokudb PARTITION  BY hash(a)  ",
```

Comment:
the case t9 is not err, others maybe have a look.

Reply:
The statement next is not correct when I make test and return error " Multiple primary key defined", but this error is returned from backend(mysql) but not radon. That why make test still all passed and does`t checkout  it.
```
create table t9(a int key, b int key, primary key(a))engine=tokudb PARTITION  BY hash(a)
```
Now  I had chang  it to ：
```
create table t9(a int unique key, b int unique key, primary key(a))engine=tokudb PARTITION  BY hash(a)
```
I had confirmed the others  does`t have the same problem.